src/SAML2/XML/ExtensionPointTrait.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+declare(strict_types=1);
+
+namespace SimpleSAML\SAML2\XML;
+
+use RuntimeException;
+use SimpleSAML\Assert\Assert;
+use SimpleSAML\XML\Exception\SchemaViolationException;
+
+/**
+ * Trait for several extension points objects.
+ *
+ * @package simplesamlphp/saml2
+ */
+trait ExtensionPointTrait
+{
+    /**
+     * Get the local name for the element's xsi:type.
+     *
+     * @return string
+     */
+    public static function getXsiTypeName(): string
+    {
+        Assert::true(
+            defined('static::NS_XSI_TYPE_NAME'),
```

Comment:
Nitpick: the constants here should probably drop the `NS_` prefix. That made sense for the other set of constants, but in this case they're just used for the XSI type.

Reply:
Will perform a search & replace here